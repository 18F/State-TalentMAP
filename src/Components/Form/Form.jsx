import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { EMPTY_FUNCTION } from '../../Constants/PropTypes';

class Form extends Component {
  constructor(props) {
    super(props);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onSubmit(e) {
    const { onSubmit } = this.props;
    // preventDefault to avoid USWDS adding query params to url
    if (e && e.preventDefault) { e.preventDefault(); }
    onSubmit(e);
  }

  render() {
    const { children, className } = this.props;
    return (
      <form className={className} onSubmit={this.onSubmit}>
        {children}
      </form>
    );
  }
}

Form.propTypes = {
  children: PropTypes.node.isRequired, // should be valid form children
  className: PropTypes.string,
  onSubmit: PropTypes.func,
};

Form.defaultProps = {
  className: '',
  onSubmit: EMPTY_FUNCTION,
};

export default Form;
