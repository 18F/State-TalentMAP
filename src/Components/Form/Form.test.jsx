import { shallow } from 'enzyme';
import React from 'react';
import toJSON from 'enzyme-to-json';
import Form from './Form';

describe('FormComponent', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(
      <Form>
        <span>child</span>
      </Form>,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
