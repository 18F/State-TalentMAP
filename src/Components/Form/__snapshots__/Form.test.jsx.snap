// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormComponent matches snapshot 1`] = `
<form
  className=""
>
  <span>
    child
  </span>
</form>
`;
